extensions/authentication-passport/src/strategy-adapter.ts

```diff
@@ -73,6 +73,18 @@ export class StrategyAdapter<U> implements AuthenticationStrategy {
         reject(new HttpErrors.InternalServerError(error));
       };
 
+      // handle redirection for oauth2 authorization flows
+      strategy.redirect = function(url: string, status: number) {
+        // set redirect options in query
+        // the controller configured with the oauth2 strategy will have to handle actual redirect
+        // the controller can access these values from the request
+        request.query['x-loopback-authentication-redirect-url'] = url;
+        request.query['x-loopback-authentication-redirect-status'] = status;
+        request.query['x-loopback-authentication-redirect-context'] = this.name;
+        // resolve with dummy user profile
+        resolve({[securityId]: ''});
```

Comment:
What does 
> resolve({[securityId]: ''}); 

do exactly; in the part it plays in the auth flow? thx

Reply:
the redirect() function cannot return a user profile ,  but the authenticate function's contract does not allow an empty resolve, I can resolve with a null value like `resolve(null)`

Reply:
I have now changes the authentication strategy to resolve either UserProfile or RedirectRoute